### DISCUSSION TITLE
Listing records with graphic accent in the name and ordering by name

### DISCUSSION THREAD

**Discussion by victorcbb** [prisma/prisma#22615] [Q&A]
### Question

I want to list records and order them by the name in ascendent order. But some of records have graphic accent in the name and the list shows in a wrong sort the records.

The list must be like:
- Alfredo
- Ângela
- Antônio
- Jota
- Victor

But i receive this:
- Alfredo
- Antônio
- Jota
- Victor
- Ângela

At the moment exists a way to fix this?

libaries versions:
prisma: 5.3.1
@prisma/client: 5.3.1

OS: pop_os 22.04
Database: MongoDB
Node: v18.17.1

**Comment by ludralph** [comment]
Hi @victorcbb 👋 

Thank you for raising this question.

We currently do have a feature request for case insensitive sorting as shown below:

- https://github.com/prisma/prisma/issues/5068

Another user also experienced the same sorting issue you've encountered and highlighted it in this [comment](https://github.com/prisma/prisma/issues/5068#issuecomment-1515179795). 

While trying to reproduce this, I noticed that using [findRaw](https://www.prisma.io/docs/orm/prisma-client/queries/raw-database-access/raw-queries#findraw) command would return the results in the sorted order you mentioned. 
For example:
```js
const res = await prisma.test.findRaw({
  options: { projection: { _id: false } },
})
```
This returned the output below which exactly matches what you expect.

```
[
  { name: 'Alfredo' },
  { name: 'Ângela' },
  { name: 'Antônio' },
  { name: 'Jota' },
  { name: 'Victor' }
]
```
If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by victorcbb** [reply]
Hi @ludralph!

Thanks for the response and sorry for the delay in getting back to you...

I researched the points you gave me. But I encountered some difficulties.

First about the filter in findRaw:

I've this collaborator model:

```JS
model Collaborator {
  id        String     @id @default(auto()) @map("_id") @db.ObjectId
  name      String
  register  String
  legal_id  String     @unique
  admission DateTime
  dismissal DateTime?
  status    Status     @default(Active)
  role      String
  document  Document[]
  organization_id String       @db.ObjectId
}
```

I tried to do a search using the filter this way but also without encapsulating the orgId in an ObjectId()

```JS
const test = await prisma.collaborator.findRaw({
      filter: { organization_id: { $eq: `ObjectId(${orgId})` } },
      options: {
        projection: { _id: false },
        sort: { name: 1 },
        limit: 10,
      },
    })

    console.log(test)
```

This way I get nothing. Removing the filter I receive the list of collaborators correctly.

The second thing is about the Prisma.JsonObject that it returns. How would I make it return a type from the Prisma.Collaborator Collaborator model?

**Reply by ludralph** [reply]
Hi @victorcbb 

The way you're trying to use ObjectId in your filter is incorrect. When using `findRaw` with MongoDB, you need to use the `$oid` operator for ObjectId fields. Here's how you should modify your query:
```js
const test = await prisma.collaborator.findRaw({
  filter: { 
    organization_id: { $eq: { $oid: orgId } }
  },
  options: {
    projection: { _id: false },
    sort: { name: 1 },
    limit: 10,
  },
})
```
`findRaw` returns a Prisma.JsonObject, which is not typed to your Prisma model. This is a known limitation of raw queries in Prisma.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
